flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -202,14 +206,18 @@ public boolean isJobManagerPortReady(Configuration config) {
                     }
                     break;
                 case LAST_STATE:
-                    final String namespace = conf.getString(KubernetesConfigOptions.NAMESPACE);
                     FlinkUtils.deleteCluster(namespace, clusterId, kubernetesClient, false);
                     break;
                 default:
                     throw new RuntimeException("Unsupported upgrade mode " + upgradeMode);
             }
         }
-        FlinkUtils.waitForClusterShutdown(kubernetesClient, conf);
+        FlinkUtils.waitForClusterShutdown(kubernetesClient, namespace, clusterId);
+        if (HighAvailabilityMode.isHighAvailabilityModeActivated(conf)) {
+            // Delete the job graph in the HA ConfigMaps so that the newly changed job config(e.g.
+            // parallelism) could take effect
+            FlinkUtils.deleteJobGraphInKubernetesHA(clusterId, namespace, kubernetesClient);
+        }
```

Comment:
Wouldn't it be more safe to execute this logic before deployment? Technically we could have an error after shutting down the job but before deleting the jobgraph.

Reply:
I think your suggestion makes sense. I will update the PR.